### DISCUSSION TITLE
Allow to extend `getNonce` to receive the connected address

### DISCUSSION THREAD

**Discussion by YasielCabrera** [rainbow-me/rainbowkit#1475] [Ideas]
Currently in order to create an Adapter with a custom nonce just by creating the `getNonce` function is enough, but in my case I have a distributed backend with a custom "getNonce" endpoint but it requires the user address in order to store the nonce in a database and later verify the message/signature and validate the nonce in a more secure way, due the distributed nature, working with session (like [this example](https://github.com/rainbow-me/rainbowkit/blob/7b31af2431cda62bf215a84b9d40fb7f66f24381/examples/with-next-siwe-iron-session/pages/api/nonce.ts)) is not too suitable for me, also other app uses the API of this backend.
Is there a way where I can use rainbowkit in a Nextjs 13 app and sign the messages in a way that when I fetch the nonce I can send the connected address?

To have something like where we can get the connected address in the `getNonce` function would be great
```ts
createAuthenticationAdapter({
  createMessage: ({ address, chainId, nonce }) => {
    // create message
  },

  getMessageBody: ({ message }) => message.prepareMessage(),

  // new
  getNonce: async ({ address }) => {
    const nonce = await getCustomNonce(address);
    if (!nonce) throw new Error();
    return nonce;
  },

  signOut: async () => {
    await signOut({ redirect: false });
  },

  verify: async ({ message, signature }) => {
    // verify...
  },
})
```

**Comment by DanielSinclair** [comment]
@YasielCabrera I see. Our SIWE adapter approach was built around the session trust model to reduce the surface area of various signature attacks. One way you could solve this is create a wrapper around `createAuthenticationAdapter` that relies on Wagmi's `useAccount()` hook to get the user address. Wagmi also keeps the active account address available at `wagmi.store` in localstorage under `state.data.account` that you can fetch in `getNonce`. Alternately, Spruce (the team that created SIWE) built a framework called [SSX](https://blog.spruceid.com/get-started-with-sign-in-with-ethereum-and-rainbowkit/) that plays well with RainbowKit to reduce the need for a SIWE backend (as they share trusted nodes). It was designed more or less for your decentralized dApp distribution approach.
